discovery-provider/src/tasks/prune_plays.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from datetime import datetime
+
+from sqlalchemy import text
+from src.tasks.celery_app import celery
+
+logger = logging.getLogger(__name__)
+
+PLAYS_ARCHIVE_TABLE_NAME = "plays_archive"
+
+# ARCHIVE_PLAYS_QUERY
+# in a sliding window, delete the oldest plays before the cutoff date
+# delete no more than MAX_BATCH
+ARCHIVE_PLAYS_QUERY = """
+    INSERT INTO
+        plays_archive
+    SELECT
+        id,
+        user_id,
+        source,
+        play_item_id,
+        created_at,
+        updated_at,
+        slot,
+        signature,
+        :current_timestamp
+    FROM
+        plays
+    WHERE
+        plays.created_at <= :cutoff_timestamp
+    ORDER BY
+        id ASC
+    LIMIT
+        :max_batch
+    """
+
+# PRUNE_PLAYS_QUERY
+# prune those same plays that were archived
+PRUNE_PLAYS_QUERY = """
```

Comment:
Another way you could do this would be as one query with a CTE:
https://www.postgresql.org/docs/9.5/sql-insert.html
https://www.postgresql.org/docs/9.6/sql-delete.html
```
with archived as (
  insert into plays_archive
  from plays
  returning plays.id
)
delete from plays
where id in (archived)
limit :max_batch
```

I recently did this in https://github.com/AudiusProject/audius-protocol/pull/2366

This would guarantee a 1-1 of ids archived <-> deleted. Although what you've written here *should* be the case, I like the extra security

Reply:
Nice! I was trying to use a CTE originally but didn't realize you could make the insert a CTE.